src/utils/nav-listener/main.cc

```diff
@@ -34,7 +34,15 @@ int main(int argc, char *argv[])
 
             while (true)
                 {
-                    udp_listener.print_content();
+                    gnss_sdr::navMsg message;
+                    if (udp_listener.read_nav_message(message))
+                        {
+                            udp_listener.print_message(message);
+                        }
+                    else
+                        {
+                            std::cout << "Error: the message cannot be parsed.\n";
```

Comment:
hi Jim, I did not change the error message, but looking at it, the conditional _is_ directly dependant on the parsing action. It would be occurring validly I'd say if the reception was ok, but the contained data was not the expected protobuf message. Of course reception errors also prevent parsing but is not the only cause. 
Am I correct here?
Maybe the output could be
` Error: the received message cannot be parsed. This can be caused by transmission errors or by unexpected message content.`

Reply:
You're right. My bad.

Personally, I prefer the reporting of errors to be at the point of detection of errors. Separating the condition from the message encourages confusion (although I don't need much help).

Reply:
no worries! I see your point.
In any case I noticed one should probably rename the function from `read_nav_message` to `receive_and_parse_nav_message`.

With the reporting where it occurs I am not completely sure. If you expect transmission errors and dropping invalid messages is ok, you might want to silently drop error messages. Moving the reporting into the read_nav_msg function would require modification of the 'library' in order to get rid of it.. (plus adding a temporary variable and a conditional jump to the function performance wise..)

Reply:
I updated the error error message and function name (didn't hear from you and interpreted the silence as no hard objectives to the suggestion, can change stuff again if requested)